src/v/cluster/controller_api.cc

```diff
@@ -307,4 +316,124 @@ ss::future<result<bool>> controller_api::are_ntps_ready(
         return !is_ready_result.has_error() && is_ready_result.value();
     });
 }
+
+ss::future<result<std::vector<partition_reconfiguration_state>>>
+controller_api::get_partitions_reconfiguration_state(
+  std::vector<model::ntp> partitions,
+  model::timeout_clock::time_point timeout) {
+    auto& updates_in_progress = _topics.local().in_progress_updates();
+
+    partitions_filter partitions_filter;
+    absl::node_hash_map<model::ntp, partition_reconfiguration_state> states;
+    for (auto& ntp : partitions) {
+        auto progress_it = updates_in_progress.find(ntp);
+        if (progress_it == updates_in_progress.end()) {
+            continue;
+        }
+        auto p_as = _topics.local().get_partition_assignment(ntp);
+        if (!p_as) {
+            continue;
+        }
+        partition_reconfiguration_state state;
+        state.ntp = ntp;
+
+        state.current_assignment = std::move(p_as->replicas);
+        state.previous_assignment = progress_it->second.get_previous_replicas();
+        state.state = progress_it->second.get_state();
+        states.emplace(ntp, std::move(state));
+
+        auto [tp_it, _] = partitions_filter.namespaces.try_emplace(
+          ntp.ns, partitions_filter::topic_map_t{});
+
+        auto [p_it, inserted] = tp_it->second.try_emplace(
+          ntp.tp.topic, partitions_filter::partitions_set_t{});
+
+        p_it->second.emplace(ntp.tp.partition);
+    }
+
+    auto result = co_await _health_monitor.local().get_cluster_health(
+      cluster_report_filter{
+        .node_report_filter
+        = node_report_filter{.ntp_filters = std::move(partitions_filter)}},
+      force_refresh::no,
+      5s + model::timeout_clock::now());
+
+    if (!result) {
+        co_return result.error();
```

Comment:
nit: we should probably just return whatever we have so far? That seems useful too.

Reply:
hmm, i am wondering if that is going to be useful, i need to think if incomplete report make sense